### PR TITLE
Add web worker support in generating account IDs and passwords

### PR DESCRIPTION
**Pull request by binoculars**
Refs: https://openscience.atlassian.net/browse/LEI-281

## Purpose
Adds web worker support for generating accounts and passwords along with a progress message that doesn't tie up the main browser threat

## Summary of changes
* Moves some functions out of the participant-creator component into the web worker
* Adds progress message (see screenshot)
* Removes bcrypt out of the account serializer in the companion PR for exp-addons

## Testing notes
When generating multiple accounts, you should see this message
screen shot 2017-01-04 at 14 51 31

### CODE REVIEW

File: app/components/participant-creator/component.js [hunk 1]
```diff
@@ -30,6 +19,8 @@ const Validations = buildValidations({
     })
 });
 
+const worker = new Worker('assets/workers/account-generator.js');
```
Comment (abought) [inline app/components/participant-creator/component.js:22]: Per MDN, it looks like we can avoid the build process issues by using blobs, though I'm on the fence as to whether it's worth it.

More info:
http://stackoverflow.com/questions/5408406/web-workers-without-a-separate-javascript-file

File: app/components/participant-creator/template.hbs [hunk 2]
```diff
@@ -1,7 +1,9 @@
 <div class="row participant-creator">
     <div class="{{if creating 'block-ui' 'block-ui-hidden'}}">
         <span class="block-ui-message">
-            Creating new records. This may take several minutes- please wait.
+            Creating new records. This may take several minutes--please wait.
+            <br>
+            Creating record {{currentRecord}} of {{batchSize}}...
```
Comment (abought) [inline app/components/participant-creator/template.hbs:6]: I like this usability improvement.

File: ember-cli-build.js [hunk 3]
```diff
@@ -2,10 +2,27 @@
...
+const Funnel = require('broccoli-funnel');
 
 module.exports = function(defaults) {
-    var app = new EmberApp(defaults, {
+    const workerDestDir = '/assets/workers';
+
+    const workers = transpiler(
+            new Funnel('app/workers', {
+            include: ['*.js'],
+            destDir: workerDestDir
```
Comment (abought) [inline ember-cli-build.js:16]: Confirmed that this worker file is found correctly, even when building in production mode (where assets would normally be fingerprinted)

File: ember-cli-build.js [hunk 4]
```diff
@@ -2,10 +2,27 @@
...
+        })
+    );
+
+    const workerDeps = new Funnel('bower_components', {
```
Comment (abought) [inline ember-cli-build.js:20]: Question: does this mean that the user is downloading bcrypt twice- once in the vendor bundle, and once as a worker dependency? No net change requested; 10 kb minified and gzipped is acceptable given what we gain.

```
> ~/code/cos/experimenter (pr/132) $ ember build --environment production

File sizes:
 - experimenter-04cc01255d49e8987fb425d8a6588606.css: 2.82 KB (1008 B gzipped)
 - experimenter-a875b0c0f0ff4c744a085d5a90304978.js: 387.41 KB (43.33 KB gzipped)
 - vendor-4615907c190f363953c9ff2bdcdacef5.js: 2.03 MB (486.76 KB gzipped)
 - vendor-9bd662c4ca09ac88c15fc65efe3e6b53.css: 166.46 KB (31.61 KB gzipped)
 - account-generator.js: 1.52 KB (720 B gzipped)
 - bcrypt.min.js: 19.44 KB (9.27 KB gzipped)
```

File: app/workers/account-generator.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        const password = bcrypt.hashSync(studyId, salt).replace('$2a$', '$2b$');
+
+        records[i] = {
+            id,
+            password,
+            extra
+        };
+    }
+
+    postMessage({records});
```
Comment (abought) [inline app/workers/account-generator.js:49]: Confirmed that the accounts generated via this mechanism can be used to login successfully on staging-isp.

**Review by abought — APPROVED** [review]
There are one or two small notes. Since this works in production builds too, I don't think the blob trick will be necessary. Moving this forward to get it in front of QA sooner. 

Can you please update the testing notes so that QA knows what and where to try this?

(I won't be updating Lookit staging/prod for a while yet)

**Comment by abought** [comment]
CI seems to be having trouble finding the new submodule commit, even after merging exp-addons/#213 . Good to go forward once tests are passing.

File: ember-cli-build.js (hunk 4 above)
Comment (binoculars) [reply ember-cli-build.js:20]: I believe so, but it would only apply to the account creation page. Seems like an acceptable tradeoff.

**Pull request merged**